test/extended/machines/cluster.go

```diff
@@ -100,4 +112,114 @@ var _ = g.Describe("[sig-node] Managed cluster", func() {
 		err := prometheus.RunQueries(context.TODO(), oc.NewPrometheusClient(context.TODO()), tests, oc)
 		o.Expect(err).NotTo(o.HaveOccurred())
 	})
+
+	g.It("should verify that nodes have no unexpected reboots [Late]", func() {
+		ctx := context.Background()
+
+		// List all nodes
+		nodes, err := oc.KubeClient().CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(nodes.Items).NotTo(o.HaveLen(0))
+
+		expected := make(map[string]int)
+		actual := make(map[string]int)
+		errs := make([]error, 0)
+
+		// Find a number of rendered-* machineconfigs for masters and workers
+		// Each rendered config adds expected reboot
+		expectedReboots := map[string]int{
+			"worker": 0,
+			"master": 0,
+		}
+		dynamicClient, err := dynamic.NewForConfig(oc.KubeFramework().ClientConfig())
+		o.Expect(err).NotTo(o.HaveOccurred())
+		machineConfigGroupVersionResource := schema.GroupVersionResource{
+			Group: "machineconfiguration.openshift.io", Version: "v1", Resource: "machineconfigs",
+		}
+		mcList, err := dynamicClient.Resource(machineConfigGroupVersionResource).List(ctx, metav1.ListOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		for i := range mcList.Items {
```

Comment:
@vrutkovs shouldnt we count the pivot reboot as well?

Reply:
Pivot reboot is counted too, as `rendered-` is generated for this (two boots, but just one recorded reboot)